File: app/src/main/java/net/steamcrafted/loadtoastlib/MainActivity.java
package net.steamcrafted.loadtoastlib;

import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;

import androidx.appcompat.app.AppCompatActivity;

import net.steamcrafted.loadtoast.LoadToast;

// Example activity
public class MainActivity extends AppCompatActivity {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final String text = "This is a long loading text...";
        final LoadToast lt = new LoadToast(this)
                .setProgressColor(Color.RED)
                .setText(text)
                .setTranslationY(100)
                .setBorderColor(Color.LTGRAY)
                .show();

        final ViewGroup root = (ViewGroup) findViewById(android.R.id.content);

        View v = new View(this);
        v.setBackgroundColor(Color.RED);

        findViewById(R.id.show).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                lt.show();
            }
        });
        findViewById(R.id.error).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                lt.error();
            }
        });
        findViewById(R.id.success).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                lt.success();
            }
        });
        findViewById(R.id.refresh).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                View v = new View(MainActivity.this);
                v.setBackgroundColor(Color.rgb((int) (Math.random() * 255), (int) (Math.random() * 255), (int) (Math.random() * 255)));
                root.addView(v, new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, 400));
            }
        });
        findViewById(R.id.hide).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                lt.hide();
            }
        });

    }
}

File: sample-app/src/main/kotlin/net/steamcrafted/loadtoastlib/MainActivity.kt
package net.steamcrafted.loadtoastlib

import android.graphics.Color
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import net.steamcrafted.loadtoast.LoadToast

// Example activity
class MainActivity : AppCompatActivity() {
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val text = "This is a long loading text..."

        val loadToast = LoadToast(this)
                .setProgressColor(Color.RED)
                .setText(text)
                .setTranslationY(100)
                .setBorderColor(Color.LTGRAY)
                .show()

        findViewById<View>(R.id.show).setOnClickListener {
            loadToast.show()
        }

        findViewById<View>(R.id.error).setOnClickListener {
            loadToast.error()
        }

        findViewById<View>(R.id.success).setOnClickListener {
            loadToast.success()
        }

        val v = View(this)
        v.setBackgroundColor(Color.RED)
        val root = findViewById<View>(android.R.id.content) as ViewGroup
        findViewById<View>(R.id.refresh).setOnClickListener {
            v.setBackgroundColor(Color.rgb((Math.random() * 255).toInt(), (Math.random() * 255).toInt(), (Math.random() * 255).toInt()))
            root.addView(v, ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, 400))
        }

        findViewById<View>(R.id.hide).setOnClickListener {
            loadToast.hide()
        }
    }
}